lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,712 @@
+# justified [(Loshchilov & Hutter, 2016)](#f4), the particular values provided here are primarily empircally driven.
+#
+# </div>
+
+
+# %%
+lr_scheduler_init = {"T_0": 1, "T_mult": 2, "eta_min": 1e-07}
+
+# %%
+# Load our lightning module...
+lightning_module_kwargs = {
+    "model_name_or_path": "microsoft/deberta-v3-base",
+    "optimizer_init": optimizer_init,
+    "lr_scheduler_init": lr_scheduler_init,
+}
+model = RteBoolqModule(**lightning_module_kwargs, experiment_tag="fts_explicit")
+
+# %% [markdown]
+# ### Callback Configuration
+#
+# The only callback required to invoke the [FinetuningScheduler](https://finetuning-scheduler.readthedocs.io/en/stable/api/finetuning_scheduler.fts.html#finetuning_scheduler.fts.FinetuningScheduler) is the [FinetuningScheduler](https://finetuning-scheduler.readthedocs.io/en/stable/api/finetuning_scheduler.fts.html#finetuning_scheduler.fts.FinetuningScheduler) callback itself.
+# Default versions of [FTSCheckpoint](https://finetuning-scheduler.readthedocs.io/en/stable/api/finetuning_scheduler.fts_supporters.html#finetuning_scheduler.fts_supporters.FTSCheckpoint) and [FTSEarlyStopping](https://finetuning-scheduler.readthedocs.io/en/stable/api/finetuning_scheduler.fts_supporters.html#finetuning_scheduler.fts_supporters.FTSEarlyStopping)
+# (if not specifying ``epoch_only_transitions``) will be included ([as discussed above](#basic_usage)) if not provided
+# in the callbacks list. For demonstration purposes I'm including example configurations of all three callbacks below.
+
+# %%
+# let's save our callback configurations for the explicit scenario since we'll be reusing the same
+# configurations for the implicit and nofts_baseline scenarios (except the  config for the
+# FinetuningScheduler callback itself of course in the case of nofts_baseline)
+earlystopping_kwargs = {"monitor": "val_loss", "min_delta": 0.001, "patience": 2}
+checkpoint_kwargs = {"monitor": "val_loss", "save_top_k": 1}
+fts_kwargs = {"max_depth": 1}
+callbacks = [
+    FinetuningScheduler(ft_schedule=ft_schedule_name, **fts_kwargs),  # type: ignore # noqa
+    FTSEarlyStopping(**earlystopping_kwargs),  # type: ignore # noqa
+    FTSCheckpoint(**checkpoint_kwargs),  # type: ignore # noqa
+]
+
+# %%
+logger = TensorBoardLogger("lightning_logs", name="fts_explicit")
+# optionally start tensorboard and monitor progress graphically while viewing multi-phase finetuning specific training
+# logs in the cell output below by uncommenting the next 2 lines
+# # %load_ext tensorboard
+# # %tensorboard --logdir lightning_logs
+# disable progress bar by default to focus on multi-phase training logs. Set to True to re-enable if desired
+enable_progress_bar = False
+
+# %%
+
+
+def train() -> None:
+    trainer = pl.Trainer(
+        enable_progress_bar=enable_progress_bar,
+        max_epochs=100,
+        precision=16,
+        strategy="dp",
+        accelerator="gpu",
+        devices=1,
```

Comment:
```suggestion
        accelerator="auto",
        devices=1 if torch.cuda.is_available() else None
```
DP with `devices=1` is just single GPU training.